single_kernel_mongo/state/charm_state.py

```diff
@@ -457,6 +458,21 @@ def set_scaling_down(self, rel_id: int, departing_unit_name: str) -> bool:
         self.unit_peer_data.update({rel_departed_key: json.dumps(scaling_down)})
         return scaling_down
 
+    def share_status_with_config_server(self, status: StatusBase):
+        """Shares this shard's status with the config server."""
+        if not self.is_role(MongoDBRoles.SHARD):
+            return
+        if not self.shard_relation:
+            return
+        if isinstance(status, ActiveStatus):
+            self.shard_state.status_ready_for_upgrade = True
+        if not isinstance(status, BlockedStatus):
+            self.shard_state.status_ready_for_upgrade = False
+        if status.message and "is not up-to date with config-server" in status.message:
+            self.shard_state.status_ready_for_upgrade = True
+
+        self.shard_state.status_ready_for_upgrade = False
```

Comment:
do we really want to overwrite the values we just set?

Reply:
Let's just say I forgot to add the returns :sweat_smile: